### PR TITLE
FOUNDATIONS: Modify Post Impressions

### LINKED ISSUE #410: FOUNDATIONS: Modify Post Impressions

### PR DESCRIPTION
**Pull request by Javlonbek-dev**
Closes #410

### CODE REVIEW

File: Taarafo.Core.Tests.Unit/Services/Foundations/PostImpressions/PostImpressionServiceTests.Exception.Modify.cs [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectPostImpressionByIdAsync(postId, profileId), Times.Once);
+                
+            this.dateTimeBrokerMock.Verify(broker =>
+                broker.GetCurrentDateTimeOffset(), Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs(expectedPostImpressionDependencyException))), Times.Once);
+
```
Comment (cjdutoit) [inline Taarafo.Core.Tests.Unit/Services/Foundations/PostImpressions/PostImpressionServiceTests.Exception.Modify.cs:114]: Remove line

File: Taarafo.Core/Services/Foundations/PostImpressions/PostImpressionService.Validations.cs [hunk 2]
```diff
@@ -113,6 +163,7 @@ private bool IsDateNotRecent(DateTimeOffset date)
             return timeDifference.Duration() > oneMinute;
         }
 
+
```
Comment (cjdutoit) [inline Taarafo.Core/Services/Foundations/PostImpressions/PostImpressionService.Validations.cs:166]: Remove extra line

File: Taarafo.Core/Services/Foundations/PostImpressions/PostImpressionService.cs [hunk 3]
```diff
@@ -53,6 +53,19 @@ public partial class PostImpressionService : IPostImpressionService
                 return maybePostImpression;
             });
 
+        public ValueTask<PostImpression> ModifyPostImpressionAsync(PostImpression postImpression) =>
+            TryCatch(async () =>
+            {
+                ValidatePostImpressionOnModify(postImpression);
+
+                var maybePostImpression =await this.storageBroker.SelectPostImpressionByIdAsync(
```
Comment (cjdutoit) [inline Taarafo.Core/Services/Foundations/PostImpressions/PostImpressionService.cs:61]: Add a space before await

**Review by cjdutoit — CHANGES_REQUESTED** [review]
Some small code clean-up tasks

File: Taarafo.Core/Services/Foundations/PostImpressions/PostImpressionService.cs (hunk 3 above)
Comment (Javlonbek-dev) [reply Taarafo.Core/Services/Foundations/PostImpressions/PostImpressionService.cs]: Done

File: Taarafo.Core/Services/Foundations/PostImpressions/PostImpressionService.Validations.cs (hunk 2 above)
Comment (Javlonbek-dev) [reply Taarafo.Core/Services/Foundations/PostImpressions/PostImpressionService.Validations.cs]: Done

File: Taarafo.Core.Tests.Unit/Services/Foundations/PostImpressions/PostImpressionServiceTests.Exception.Modify.cs (hunk 1 above)
Comment (Javlonbek-dev) [reply Taarafo.Core.Tests.Unit/Services/Foundations/PostImpressions/PostImpressionServiceTests.Exception.Modify.cs]: Done

File: Taarafo.Core/Services/Foundations/PostImpressions/PostImpressionService.Validations.cs [hunk 4]
```diff
@@ -33,6 +35,45 @@ private void ValidatePostImpressionOnAdd(PostImpression postImpression)
                 (Rule: IsNotRecent(postImpression.CreatedDate), Parameter: nameof(PostImpression.CreatedDate)));
         }
 
+        private void ValidatePostImpressionOnModify(PostImpression postImpression)
+        {
+            ValidatePostImpressionIsNotNull(postImpression);
+
+            Validate(
+            (Rule: IsInvalid(postImpression.PostId), Parameter: nameof(PostImpression.PostId)),
```
Comment (cjdutoit) [inline Taarafo.Core/Services/Foundations/PostImpressions/PostImpressionService.Validations.cs:43]: Should TAB in one more for all the rules.

**Review by cjdutoit — CHANGES_REQUESTED** [review]
Fix formatting of modify rules

File: Taarafo.Core/Services/Foundations/PostImpressions/PostImpressionService.Validations.cs (hunk 4 above)
Comment (Javlonbek-dev) [reply Taarafo.Core/Services/Foundations/PostImpressions/PostImpressionService.Validations.cs]: Done

**Review by cjdutoit — APPROVED** [review]
Fix formatting of modify rules

**Pull request merged**